src/cli/go.js

```diff
@@ -11,22 +11,24 @@ function die(std, message) {
   return 1;
 }
 
-const noLoadArg = "--no-load";
+const NO_LOAD_ARG = "--no-load";
 
 const goCommand: Command = async (args, std) => {
+  let noLoad = false;
+  if (args.length === 1 && args[0] === NO_LOAD_ARG) {
+    noLoad = true;
+  } else if (args.length !== 0) {
+    return die(std, "usage: sourcecred go [--no-load]");
+  }
+
   const commandSequence = [
     {name: "load", command: load},
     {name: "graph", command: graph},
     {name: "score", command: score},
   ];
 
-  if (args.length === 1 && args[0] === noLoadArg) {
-    commandSequence.shift();
-  } else if (args.length !== 0) {
-    return die(std, "usage: sourcecred go [--no-load]");
-  }
-
   for (const {name, command} of commandSequence) {
+    if (name === "load" && noLoad) continue;
```

Comment:
I like this better than the previous version, because it’s more explicit
about the branching. IMHO even clearer would be something like:

```javascript
const commandSequence = [];
if (!noLoad) {
  commandSequence.push({name: "load", command: load});
}
commandSequence.push({name: "graph", command: graph});
commandSequence.push({name: "score", command: score});
```

because `commandSequence` actually corresponds to the sequence of
commands that will be executed.

(Also, it would probably be clearer to call the variable `shouldLoad`
rather than `noLoad`, since `if (shouldLoad)` is slightly easier to
understand than `if (!noLoad)`.)